### PR TITLE
feat: `to_iceberg` support for filling missing columns in the DataFrame with None

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Feature or Bugfix
- Feature

### Detail
- If `to_iceberg` is invoked with `fill_missing_columns_in_df=True`, missing columns in the DataFrame will be filled in with None

### Relates
- #2458

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by malachi-constant — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 80ce9f3dceb3ae23248153664dadba1746e1cbb1
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f8e41937-ebd5-4abb-ab7c-02902236a712%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 80ce9f3dceb3ae23248153664dadba1746e1cbb1
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6481ecf8-6c83-459d-8cc6-4783b2507179%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/athena/_write_iceberg.py [hunk 1]
```diff
@@ -269,6 +274,10 @@ def to_iceberg(
         If none is provided, the AWS account ID is used by default
     schema_evolution: bool
         If True allows schema evolution for new columns or changes in column types.
+        Missing columns will throw an error unless ``schema_fill_missing`` is set to ``True``.
+    schema_fill_missing: bool
+        If True, fill missing columns with NULL values.
+        Only takes effect if ``schema_evolution`` is set to True.
```
Comment (kukushking) [inline awswrangler/athena/_write_iceberg.py:280]: Do I understand correctly that if `schema_evolution=True` and this is `False`, then the table will be updated such that columns missing in the df are dropped from the table, and if `True`, they will be added and filled with `None`?

If so, this request, together with #2606 probably makes the case for `mode={append, overwrite}` kwarg, similar to the  `wr.s3.to_parquet` i.e. if the mode is `append` the data will be added to the table (and missing columns filled with `NULL`), and `overwrite` would overwrite the data and update the schema. Wyt?

Comment (LeonLuttenberger) [reply awswrangler/athena/_write_iceberg.py:280]: Just to make sure we're on the same page, your proposal is for the behavior to be the following, if a DataFrame came in that didn't match the current Iceberg schema:
1. `schema_evolution=False` and `mode="append"` - throw error saying schema doesn't match
2. `schema_evolution=False` and `mode="overwrite"` - just overwrite the entire table
3. `schema_evolution=True` and `mode="append"` - make the required changes to the schema
4. `schema_evolution=True` and `mode="overwrite"` - same behavior as 2.

The only thing I'm not sure of is what happens under scenario 1 if the only difference between the two schemas is that the DF is missing certain columns. Do we just fill them with None, or throw an error? I believe the former is the behavior for `wr.s3.to_parquet`.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: a432378083905ea3e5611bc5b144ab69d6170738
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c01a89c4-b430-48a8-aac6-989a281b21e2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: a432378083905ea3e5611bc5b144ab69d6170738
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7af0534b-36d4-4e98-990e-ec36bfe984d7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/athena/_write_iceberg.py (hunk 1 above)
Comment (Mroq93) [reply awswrangler/athena/_write_iceberg.py:280]: Hi,
Please allow me to comment.
I was trying to save pyspark DF to Iceberg table.
https://iceberg.apache.org/docs/latest/spark-writes/#writing-with-dataframes

`df.writeTo(path).option("mergeSchema","true").append()`

I was checking 2 scenarios:

- Saving DF with missing _column_a_  - result is that missing _column_a_ was filled with `null` value in the table
- Saving DF with the new additional _column_b_ - result is that table has new additional _column_b_, but the previous data set is filled with `null` for that column

I use `wr.s3.to_parquet` also and the behavior is the same as I described above.

I would expect that `to_iceberg` will return the same results with needed configuration setup.

**Review by kukushking — APPROVED** [review]
Approved as discussed internally.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b9515cc1e5a415f48db55fd440256945c17f23ae
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=da4227bf-287f-41ae-b8d4-e19e644afcb3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: b9515cc1e5a415f48db55fd440256945c17f23ae
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a414a588-e9d6-4f8f-a827-e2d29bff1e6e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 71b778cc1abf326d9509950f54dc1c60f0b004aa
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5a935ce8-0d5f-434f-8667-e797f83b84af%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 71b778cc1abf326d9509950f54dc1c60f0b004aa
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1a7df28e-a15b-4543-aeee-a6a6f20736ad%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**